tests/test_cmdLineFwk.py

```diff
@@ -66,6 +67,23 @@ def makeTmpFile(contents=None):
         os.remove(tmpname)
 
 
+@contextlib.contextmanager
+def makeSQLiteRegistry():
+    """Context manager to create new empty registry database.
+
+    Yields
+    ------
+    config : `RegistryConfig`
+        Registry configuration for initialized registry database.
+    """
+    with makeTmpFile() as filename:
+        uri = f"sqlite:///{filename}"
```

Comment:
Does this have to be a file? Can't it be an in-memory sqlite?

Reply:
In-memory does not work in this case, there is a discussion on JIRA about this issue. Basically if you want to open registry in read-only mode then you can't use in-memory because every session for in-memory sqlite makes it's own database which cannot be shared, and read-only database cannot be initialized with all the stuff that we need.